corehq/apps/hqwebapp/tasks.py

```diff
@@ -43,8 +44,8 @@ def mark_subevent_gateway_error(messaging_event_id, error, retrying=False):
 
 @task(serializer='pickle', queue="email_queue",
       bind=True, default_retry_delay=15 * 60, max_retries=10, acks_late=True)
-def send_mail_async(self, subject, message, from_email, recipient_list,
-                    messaging_event_id=None, domain=None):
+def send_mail_async(self, subject, message, configuration: EmailConfigurationManager,
```

Comment:
Thanks for pointing out the celery issue - I wasn't aware of that. The only other option I can think of is having the caller decompose the properties and pass the arguments individually to `send_mail_async`. Or is there is a better option?

> It looks like this arg is passed in to allow control over which gateway - why not just switch everything to the custom gateway when configured to the domain? Is it to do a phased rollout or something?

I wanted to limit this PR to only conditional alerts and broadcast as that was the scope of the ticket. I haven't looked into each of the callers of `send_mail_async` to know what it would impact, I plan to do that in the ticket to implement this to all system emails. But I don't think all emails would use the custom gateway. For example, I think emails involved with creating a Commcare account uses `send_mail_async` and we would to use the default Gateway for that.

Reply:
That makes sense - I do think the eventual outcome is that any emails tagged with a domain will use the domain's custom gateway, but it's certainly fair to be cautious about that.

> The only other option I can think of is having the caller decompose the properties and pass the arguments individually to send_mail_async. Or is there is a better option?

Another option is to pass in an arg like `use_domain_gateway`, and then in the body of this function do something like:
```python
if use_domain_gateway and domain_gateway_exists:
    configuration = getEmailConfiguration
else:
    configuration = DefaultEmailConfiguration
```
Nice thing about that is it requires callers of this function to have less knowledge of how this works (and fewer imports)

Reply:
Thanks for the suggestion, i like this method
[37107ae](https://github.com/dimagi/commcare-hq/pull/33484/commits/37107aedbcdf333864073bd984334240d03a48b7)